src/main/java/core/AbstractGameState.java

```diff
@@ -153,9 +162,35 @@ public final AbstractGameState copy(int playerId) {
 
     /* Methods to be implemented by subclass, protected access. */
 
+    public IExtendedSequence currentActionInProgress() {
+        return actionsInProgress.isEmpty() ? null : actionsInProgress.peek();
+    }
+
+    public boolean isActionInProgress() {
+        checkActionsInProgress();
+        return !actionsInProgress.empty();
+    }
+
+    public void setActionInProgress(IExtendedSequence action) {
+        if (gamePhase != DominionGameState.DominionGamePhase.Play)
```

Comment:
Dominion-specific shouldn't be in core framework.

Reply:
Indeed it should not! Thanks for catching that.